### ISSUE TITLE
Release alpha version for 3.0 alphas

### ISSUE THREAD

**Issue by JakeWharton** [JakeWharton/butterknife#999]
With proper feature plugin support

**Comment by c2c2** [comment]
I want use in feature plugin but show an error:
Attribute value must be constant
and i can't use it

**Comment by elodieferrais** [comment]
I faced the error mentioned in this issue [963](https://github.com/JakeWharton/butterknife/issues/963). 
I am now using a [forked version of the repo](https://github.com/dotloop/butterknife).
The main changes are: 
- [the modification of the gradle distribution url](https://github.com/dotloop/butterknife/commit/31752f7dda9a2f751b495be4db18a94041eb55a1#diff-fd9d61e37d939c941e7a62077f38d62d) 
- [the update of the androidPlugin ](https://github.com/dotloop/butterknife/commit/31752f7dda9a2f751b495be4db18a94041eb55a1#diff-c197962302397baf3a4cc36463dce5ea)
- [the addition of one line to generate R2 in feature module](https://github.com/dotloop/butterknife/commit/f2bb146b918bbeef65307cdc3db953ea06cac429#diff-5845b6382cc7e372f37cd06161fbc3c3
) 

I don't see the error mentioned in the issue #963 anymore but it still doesn't assemble.
The error I see is 
```
error: cannot find symbol
import com.elodieferrais.myapp.R2;
symbol: class R2
location: package com.elodieferrais.myapp
```

What I don't understand is that I see the R2 file being generated but for some reason the java class doesn't pick it up... 
Like it has been generated but too late maybe... ¯\_(ツ)_/¯

I created a sample app where I reproduced the error: https://github.com/elodieferrais/ButterknifeSample

The title and comment of this ticket seemed to be similar to the problem I am facing, did you create it to fix an issue like this one?

**Comment by SmallerRiver** [comment]
I added all the dependencies but there were errors
Error:(2,0) Cannot add extension with name "buildOutputs" ,as there is an extension already registered with that name
I use AS3.0 canary 4

**Comment by Yuliya-Kaleda** [comment]
AS 3.0 Canary 5
Butterknife 8.7.0

```
Caused by: java.lang.NoSuchMethodError: com.android.build.gradle.api.BaseVariant.getOutputs()Ljava/util/List;
        at butterknife.plugin.ButterKnifePlugin$applyPlugin$1.execute(ButterKnifePlugin.kt:27)
        at butterknife.plugin.ButterKnifePlugin$applyPlugin$1.execute(ButterKnifePlugin.kt:15)
        at org.gradle.internal.ImmutableActionSet$CompositeSet.execute(ImmutableActionSet.java:196)
        at org.gradle.internal.MutableActionSet.execute(MutableActionSet.java:35)
        at org.gradle.api.internal.DefaultDomainObjectCollection.doAdd(DefaultDomainObjectCollection.java:204)
        at org.gradle.api.internal.DefaultDomainObjectCollection.add(DefaultDomainObjectCollection.java:198)
        at com.android.build.gradle.LibraryExtension.addVariant(LibraryExtension.java:70)
        at com.android.build.gradle.FeatureExtension.addVariant(FeatureExtension.java:80)
        at com.android.build.gradle.internal.ApiObjectFactory.create(ApiObjectFactory.java:137)
        at com.android.build.gradle.BasePlugin.lambda$createAndroidTasks$8(BasePlugin.java:625)
        at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:81)
        at com.android.build.gradle.BasePlugin.createAndroidTasks(BasePlugin.java:610)
        at com.android.build.gradle.BasePlugin.lambda$null$6(BasePlugin.java:537)
        at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:81)
        at com.android.build.gradle.BasePlugin.lambda$createTasks$7(BasePlugin.java:533)
        at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:91)
        at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:80)
        at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:42)
        at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:230)
        at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:149)
        at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:58)
        ... 90 more


```

**Comment by elodieferrais** [comment]
@Yuliya-Kaleda The stacktrace you posted is related to the issue [963](https://github.com/JakeWharton/butterknife/issues/963) I believe.

**Comment by Yuliya-Kaleda** [comment]
@elodieferrais it is. You added a link from [963 ](https://github.com/JakeWharton/butterknife/issues/963) to this issue. So I am posting here to inform that feature plugin is not fully supported.

**Comment by diswy** [comment]
use 8.5.1 can work it.

**Comment by jaredsburrows** [comment]
@JakeWharton You can close this now as you have `9.0.0-SNAPSHOT` released.

**Comment by phileo** [comment]
Is support for "api" and "implementation" separation to replace "compile" within scope of this request?
https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation

**Comment by lowwor** [comment]
Since AS 3.0 becomes stable, when migrating our project to AS 3.0, can we use 9.0.0-SNAPSHOT in production, is it stable enough? If not, can we have a stable mid version before 9.0.0 release. Thanks

**Comment by p-fischer** [comment]
This issue is resolved with the release of `9.0.0-rc1`.
